app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -214,6 +215,41 @@ class FireDatabase : Database() {
         return updateUserData(dataUpdated)
     }
 
+    override fun addUserToFriendsList(userId: String): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        isUserStoredInDatabase(userId).thenAccept {
+            if(!it){
+                future.completeExceptionally(java.lang.Error("The user with $userId is not present on the database."))
+            }else{
+                val newFriend = HashMap<String,Any?>()
+                newFriend.put(userId, null)
+                accessUserAccountFile(userId).child(friendsListFile).updateChildren(newFriend)
+                    .addOnSuccessListener { future.complete(true) }
+                    .addOnFailureListener { future.completeExceptionally(it) }
+            }
+        }
+        return future
+    }
+
+    override fun removeUserToFriendsList(userId: String): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+
+        accessUserAccountFile(userId).child(friendsListFile).get()
+            .addOnSuccessListener {previousFriendsList->
+                if(!previousFriendsList.children.contains(userId as DataSnapshot)){
+                    future.completeExceptionally(java.lang.Error("The userId $userId is not in the friends list."))
+                }else{
+                    //delete the user if it is present in the friends list
+                    val updateData = previousFriendsList.children.filter { (it.key as String)!=userId }
+                    accessUserAccountFile(userId).child(userId).setValue(updateData)
```

Comment:
I don't think it's supposed to access `users/$uid/$uid`.

Reply:
Thanks for remarking the typo